webapp/src/compiler.ts

```diff
@@ -718,11 +718,24 @@ function upgradeFromBlocksAsync(): Promise<UpgradeResult> {
 
             const xml = Blockly.Xml.textToDom(text);
             pxt.blocks.domToWorkspaceNoEvents(xml, ws);
-            patchedFiles["main.blocks"] = text;
+            pxtblockly.upgradeTilemapsInWorkspace(ws, pxt.react.getTilemapProject());
+            const upgradedXml = Blockly.Xml.workspaceToDom(ws);
+            patchedFiles["main.blocks"] = Blockly.Xml.domToText(upgradedXml);
+
             return pxt.blocks.compileAsync(ws, info)
         })
         .then(res => {
             patchedFiles["main.ts"] = res.source;
+            return project.buildTilemapsAsync();
+        })
+        .then(() => {
+            const compiledFiles = project.getAllFiles();
+            const generatedFileNames = Object.keys(compiledFiles).filter(el => /\.g\.ts/.test(el));
```

Comment:
why not just copy over all non-main.ts files?

Reply:
No particular reason, these were just the ones that were breaking things.

I'll switch it so it hopefully avoids the issue in the future~